test/gtest/type-builder.cpp

```diff
@@ -101,6 +102,47 @@ TEST_F(TypeTest, TypeIterator) {
   EXPECT_EQ(reverse, tuple.rend());
 }
 
+TEST_F(TypeTest, PrintTypes) {
+  TypeBuilder builder(4);
+
+  Type refStructA = builder.getTempRefType(builder[0], Nullable);
+  Type refStructB = builder.getTempRefType(builder[1], Nullable);
+  Type refArrayA = builder.getTempRefType(builder[2], Nullable);
+  Type refArrayB = builder.getTempRefType(builder[3], Nullable);
+  builder[0] = Struct({Field(refArrayB, Immutable)});
+  builder[1] = Struct({Field(refStructA, Immutable)});
+  builder[2] = Array(Field(refStructB, Immutable));
+  builder[3] = Array(Field(refArrayA, Immutable));
+
+  auto result = builder.build();
+  ASSERT_TRUE(result);
+  auto built = *result;
+
+  std::vector<HeapType> structs{built[0], built[1]};
+  std::vector<HeapType> arrays{built[2], built[3]};
+
+  using ArrayPrinter = IndexedTypeNameGenerator<DefaultTypeNameGenerator>;
+  ArrayPrinter printArrays(arrays, "array");
+  using StructPrinter = IndexedTypeNameGenerator<ArrayPrinter>;
+  StructPrinter print(structs, printArrays, "struct");
```

Comment:
I not sure how to read the intent of this code. Why do we need a special array and  struct printers? Why are those types not printed by the default printer?

Reply:
Yeah the names here are very similar to the names produced by the default printer, but the point of the test is to test `IndexedTypeNameGenerator` specifically. I updated the name of the test and added a comment.